spec/javascripts/Users/components/PermissionsForm.spec.jsx

```diff
@@ -0,0 +1,244 @@
+    expect(wrapper.containsAllMatchingElements([
+      <input id={'user_member_permission_ids_portal'} />,
+      <input id={'user_member_permission_ids_finance'} />,
+      <input id={'user_member_permission_ids_settings'} />
+    ])).toBe(true)
+
+    // $FlowFixMe: waiting for https://github.com/facebook/flow/pull/7298 to be merged
+    expect(wrapper.containsAllMatchingElements([
+      <input id={'user_member_permission_ids_partners'} />,
+      <input id={'user_member_permission_ids_monitoring'} />,
+      <input id={'user_member_permission_ids_plans'} />
+    ])).not.toBe(true)
+
+    wrapper.setProps({ features: ['partners', 'monitoring', 'plans'] })
+
+    // $FlowFixMe: waiting for https://github.com/facebook/flow/pull/7298 to be merged
+    expect(wrapper.containsAllMatchingElements([
+      <input id={'user_member_permission_ids_partners'} />,
+      <input id={'user_member_permission_ids_monitoring'} />,
+      <input id={'user_member_permission_ids_plans'} />
+    ])).toBe(true)
+
+    // $FlowFixMe: waiting for https://github.com/facebook/flow/pull/7298 to be merged
+    expect(wrapper.containsAllMatchingElements([
+      <input id={'user_member_permission_ids_portal'} />,
+      <input id={'user_member_permission_ids_finance'} />,
+      <input id={'user_member_permission_ids_settings'} />
+    ])).not.toBe(true)
+  })
+
+  /* TODO: Not sure if should throw an error */
+  it('should throw an error if a feature is not known', () => {
+    const test = () => wrapper.setProps({ features: ['Some Random Feature'] })
+    expect(test).toThrowError()
+  })
+
+  it('should have no checked features by default', () => {
+    const features = ['portal', 'finance', 'settings']
+    wrapper.setProps({ features })
+    expect(wrapper.state('checkedFeatures')).toEqual([])
+
+    expect(wrapper.find('.user_member_permission_ids').length).toBeGreaterThan(0)
+    expect(wrapper.find('.user_member_permission_ids').some('[checked="true"]')).toBe(false)
+  })
+
+  it('should be able to select features', () => {
+    const features = ['portal', 'finance', 'settings']
+    wrapper.setProps({ features })
+    wrapper.setState({ checkedFeatures: [] })
+
+    wrapper.find('.user_member_permission_ids').first().simulate('change')
+    expect(wrapper.state('checkedFeatures')).toHaveLength(1)
+
+    wrapper.find('.user_member_permission_ids').first().simulate('change')
+    expect(wrapper.state('checkedFeatures')).toHaveLength(0)
+
+    wrapper.find('.user_member_permission_ids').at(1).simulate('change')
+    expect(wrapper.state('checkedFeatures')).toHaveLength(1)
+    wrapper.find('.user_member_permission_ids').at(2).simulate('change')
+    expect(wrapper.state('checkedFeatures')).toHaveLength(2)
+    wrapper.find('.user_member_permission_ids').at(3).simulate('change')
+    expect(wrapper.state('checkedFeatures')).toHaveLength(3)
+  })
+
+  it.skip('should render a ServiceFeatureAccess if any feature granting service access is checked', () => {
```

Comment:
Why skip?

Reply:
Well the spec is not yet implemented, it's a to-do for a future PR.